src/tests/JIT/Regression/JitBlue/Runtime_76194/Runtime_76194.cs

```diff
@@ -0,0 +1,168 @@
+
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static int Main()
+    {
+        if (!OperatingSystem.IsWindows() && !OperatingSystem.IsLinux())
+        {
+            return 100;
+        }
+        
+        for (int i = 0; i < 100; i++)
+        {
+            using CrossVirtualAlloc alloc = new();
+            if (alloc.IsFailedToCommit)
+            {
+                Console.WriteLine("VirtualAlloc/mmap/mprotect failed, giving up on test.");
+                break;
+            }
+            Read1(alloc.GetPointerNearPageEndFor<Data1>());
+            Write1(alloc.GetPointerNearPageEndFor<Data1>(), default);
+            Read2(alloc.GetPointerNearPageEndFor<Data2>());
+            Write2(alloc.GetPointerNearPageEndFor<Data2>(), default);
+            Read3(alloc.GetPointerNearPageEndFor<Data3>());
+            Write3(alloc.GetPointerNearPageEndFor<Data3>(), default);
+            Read4(alloc.GetPointerNearPageEndFor<Data4>());
+            Write4(alloc.GetPointerNearPageEndFor<Data4>(), default);
+            Read5(alloc.GetPointerNearPageEndFor<Data5>());
+            Write5(alloc.GetPointerNearPageEndFor<Data5>(), default);
+        }
+        return 100;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static T Read<T>(byte* location) => Unsafe.ReadUnaligned<T>(location);
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static void Write<T>(byte* location, T t) => Unsafe.WriteUnaligned(location, t);
+}
+
+// Cross-platform implementation of VirtualAlloc that is focused on reproducing problems
+// where JIT emits code that reads/writes more than requested
+public unsafe class CrossVirtualAlloc : IDisposable
+{
+    private readonly byte* _ptr;
+
+    public CrossVirtualAlloc()
+    {
+        if (OperatingSystem.IsWindows())
+        {
+            const int MEM_COMMIT = 0x1000;
+            const int MEM_RESERVE = 0x2000;
+            const int PAGE_READWRITE = 0x04;
+            const int MEM_RELEASE = 0x8000;
+
+            byte* reservePtr = VirtualAlloc(null, PageSize * 2, MEM_RESERVE, PAGE_READWRITE);
+            if (reservePtr != null)
+            {
+                _ptr = VirtualAlloc(reservePtr, PageSize, MEM_COMMIT, PAGE_READWRITE);
+                if (_ptr == null)
+                {
+                    VirtualFree(reservePtr, 0, MEM_RELEASE);
+                }
+            }
+        }
+        else
+        {
+            const int PROT_NONE = 0x0;
+            const int PROT_READ = 0x1;
+            const int PROT_WRITE = 0x2;
+            const int MAP_PRIVATE = 0x02;
+            const int MAP_ANONYMOUS = 0x20;
+
+            _ptr = mmap(null, 2 * PageSize, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+            if (_ptr != null && mprotect(_ptr + PageSize, PageSize, PROT_NONE) != 0)
+            {
+                munmap(_ptr, 2 * PageSize);
```

Comment:
This looks like it will result in a double free

Reply:
(But I doubt that's the cause of #78758)

Reply:
Ah, I see the reason now:
```scala
On success, mmap() returns a pointer to the mapped area.  On
       error, the value MAP_FAILED (that is, (void *) -1) is returned,
       and [errno](https://man7.org/linux/man-pages/man3/errno.3.html) is set to indicate the error.
```

So this does not correctly check that we failed to allocate the memory.